nixos/modules/virtualisation/amazon-init.nix

```diff
@@ -43,7 +41,18 @@ let
     nixos-rebuild switch
   '';
 in {
-  boot.postBootCommands = ''
-    ${bootScript} &
-  '';
+  systemd.services.amazon-init = {
+    inherit script;
+    description = "Reconfigure the system from EC2 userdata on startup";
+
+    wantedBy = [ "sshd.service" ];
+    after = [ "network-online.target" ];
+    requires = [ "network-online.target" ];
+ 
+    restartIfChanged = false;
+    unitConfig.X-StopOnRemoval = false;
+
+    serviceConfig.Type = "oneshot";
```

Comment:
Once per boot I think is the most sensible thing. Unclear to me how `RemainAfterExit` will play with the `nixos-rebuild switch` in between, but I guess if the config for this service doesn't change due to `nixos-rebuild switch`, then perhaps `RemainAfterExit` will make a difference?